src/Pages/Withdrawals.razor

```diff
@@ -965,64 +985,72 @@
+                newRequest.Wallet = await WalletRepository.GetById(newRequest.WalletId);
+                    
+                // Use either the custom fee rate or the recommended fee rate based on mempool fee type
+                decimal newFeeRate;
+                if (newRequest.MempoolRecommendedFeesType == MempoolRecommendedFeesType.CustomFee && newRequest.CustomFeeRate.HasValue)
+                {
+                    newFeeRate = (decimal)newRequest.CustomFeeRate.Value;
+                }
+                else
+                {
+                    // Use the recommended fee rate from NBXplorer service based on the selected mempool fee type
+                    var recommendedFeeRate = await GetRecommendedFeeRate(newRequest.MempoolRecommendedFeesType);
+                    newFeeRate = (decimal)recommendedFeeRate;
+                }
 
-            // If the new fee plus the amount exceeds the utxo sum value, show an error
-            // NOTE: revisit this when full rbf because we could add a UTXO
-            var txInfo = await NBXplorerService.GetTransactionAsync(uint256.Parse(_selectedRequest.TxId));
-            var tx = txInfo.Transaction;
-            long vSize = tx.GetVirtualSize();
-            decimal newFee = (vSize * newFeeRate) / 100_000_000m;
+                // If the new fee plus the amount exceeds the utxo sum value, show an error
+                // NOTE: revisit this when full rbf because we could add a UTXO
+                var txInfo = await NBXplorerService.GetTransactionAsync(uint256.Parse(_selectedRequest.TxId));
+                var tx = txInfo.Transaction;
+                long vSize = tx.GetVirtualSize();
+                decimal newFee = (vSize * newFeeRate) / 100_000_000m;
 
-            // Retrieve from database to get the UTXOs
-            _selectedRequest = await WalletWithdrawalRequestRepository.GetById(_selectedRequest.Id);
+                // Retrieve from database to get the UTXOs
+                _selectedRequest = await WalletWithdrawalRequestRepository.GetById(_selectedRequest.Id);
 
-            if (_selectedRequest == null)
-            {
-                ToastService.ShowError("Withdrawal request not found");
-                return;
-            }
+                if (_selectedRequest == null)
+                {
+                    ToastService.ShowError("Withdrawal request not found");
+                    return;
+                }
 
-            if (_selectedRequest.UTXOs != null && _selectedRequest.UTXOs.Count > 0)
-            {
-                var inputAmount = Money.Satoshis(_selectedRequest.UTXOs.Sum(x => x.SatsAmount)).ToDecimal(MoneyUnit.BTC); // Convert to BTC
-                if (inputAmount <= newFee + _selectedRequest.TotalAmount + Constants.BITCOIN_DUST.ToDecimal(MoneyUnit.BTC))
+                if (_selectedRequest.UTXOs != null && _selectedRequest.UTXOs.Count > 0 && _selectedRequest.WalletWithdrawalRequestDestinations!.Count > 1)
```

Comment:
I would add a single line comment explaining that this enables changeless for 1 output txs

Reply:
done!